File: 1.Domain/Domain.Model/PreOffer.cs
using Core;
using Domain.Model.Enum;
using System;

namespace Domain.Model
{
    public class PreOffer: Entity<int>
    {                
        public DateTime? PreOfferDate { get; set; }
        public float Salary { get; set; }
        public PreOfferStatus Status { get; set; }
        public int VacationDays { get; set; }
        public HealthInsuranceEnum HealthInsurance { get; set; }
        public int ProcessId { get; set; }
        public int Bonus { get; set; }
        public string Notes { get; set; }
        public DateTime TentativeStartDate { get; set; }
    }
}

File: src/entities/pre-offer.ts
import { preOfferStatusEnum } from './enums/pre-offer-status.enum';
import { HealthInsuranceEnum } from './enums/health-insurance.enum';
import { TestBed } from '@angular/core/testing';


export class PreOffer {
    id: number;
    preOfferDate: Date;
    salary: number;
    status: preOfferStatusEnum;
    processId: number;
    healthInsurance: HealthInsuranceEnum;
    vacationDays: number;
    bonus: number;
    notes: String;
    tentativeStartDate: Date;
    healthInsuranceText: string; // ??
}


// export class PreOffer {
//     id: number;
//     preOfferDate: Date;
//     grossSalary: number;
//     vacationDays: number;
//     healthInsurance: HealthInsuranceEnum;
//     healthInsuranceText: string;
//     
//     condition: preOfferStatusEnum;
//     bonus: text;
//     processId: number;
// }
